syntax = "proto3";

import "google/protobuf/empty.proto";

package code_executor;

message CodeExecutionRequest {
  string code = 1;
}

message CodeExecutionResponse {
  string output = 1;
  string error = 2;
  int32 exit_code = 3;
}

message GetToolListResponse {
  repeated Tool tools = 1;
}

message Tool {
  string name = 1;
  string description = 2;
  string output_type = 3;
  map<string, ToolInput> inputs = 4;
}

message ToolInput {
  string type = 1;
  string description = 2;
}

service CodeExecutor {
  rpc ExecuteCode(CodeExecutionRequest) returns (CodeExecutionResponse);
  rpc GetToolList(google.protobuf.Empty) returns (GetToolListResponse);
}